bootloader/hos/hos.c

```diff
@@ -644,7 +644,7 @@ int hos_launch(ini_sec_t *cfg)
 	_free_launch_components(&ctxt);
 
 	// Copy BCT if debug mode is enabled.
-	if (ctxt.pkg1_id->kb != KB_FIRMWARE_VERSION_600)
+	if (ctxt.pkg1_id->kb < KB_FIRMWARE_VERSION_600)
 		memset((void *)0x4003D000, 0, 0x3000);
```

Comment:
Now the else clause, for 6.0.0 and up.

Reply:
What would this ``else`` clause do? (I don't have a copy of nxbootloader for 6, or anything else, for that matter.)

Reply:
Oh, turns out I have nxbootloader!
And yep, I have IDA Pro...

What do I need to do?

Reply:
Normally, the BCT is copied from x40008000. Search for that (or in decimal).
You will see that they have a new address and a new size.

You will not memset that addr though with that size, because you are going to hit a memory hole and hang. If you do this: secmon_addr + secmon_size + warmboot_size and then align it to x100, you will get the new addr that needs to be memset with the new size.

Reply:
I don't see a size being in any of these occurrences...
https://there-are-no-underscores-here-but-this-is-still-an.underscore.party/byZ83ZpC.png

Please correct me if I'm doing this wrong

Reply:
load the decrypted pkg1 to x40001000. Then try to search x40008000

Reply:
still no...
what should i be using to search?

Reply:
First set the arch to ARM LE and then auto analyze it.
Then search for Data and give the 0x40008000 value.

Reply:
It's the immediate value

Reply:
Any news on that?

Reply:
I'm too dumb to figure this out, I'm so sorry.
I can't figure out where to look...

Reply:
You came so far. You need to do a little research.

Here's the thing:
You do the memset and I'll accept the merge.

------

So, with IDA when you drop a binary you have to choose it's architecture by selecting the correct processor type. That's because it's a binary and it does not have any relevant info.

Tegra X1 has a boot and power management CPU.
That CPU is an arm7tdmi-s. So it's a an *arm* core with *Little-Endian* (LE) byte order.
Additionally it has support for a max of *32bit* address space.

pkg1 is *loaded* at 0x40010000.

Now because this a binary and not a normal executable, it does not have any classic info at header that can be parsed by IDA.
So you have to *select* the whole screen there and tell IDA that it is *Code*.
Ofc ida will ask you if it is all of it code and will offer to analyze it.

After that you need to search for an *immediate value*. That value is 0x40008000.

Because it needs a lot of practice to read asm, to easily understand it, you have to convert it to *pseudo-code*.
Start searching until you found something relevant that has 0x40008000.